### PR TITLE
Generic implementations

### PR DESCRIPTION
**Pull request by jkatzy**

### CODE REVIEW

File: Generic_implementations.tex [hunk 1]
```diff
@@ -0,0 +1,106 @@
+\subsection{Implementations using generic distributed frameworks}
+Distributed system have already been implemented in the industry to solve the
+problem of companies possessing massive amounts of data that they want to be able
+to query and analyze. These frameworks largely rely on the fact that it is cheaper
+to have multiple servers with relatively small storage capacity each in clusters,
```
Comment (jverbraeken) [inline Generic_implementations.tex:5]: with a relatively small storage capacity and computational power

**Review by jverbraeken — CHANGES_REQUESTED** [review]
Generally looking good. However, I'm not sure if GFS is the main basis of existing Distributed Machine Learning frameworks. GFS is very much not open source, so that suggest that only Google could use Distributed Machine Learning.
Spark is definitely something that is used a lot in Distributed Machine Learning, however, Spark in itself cannot do any Machine Learning whatsoever.
According to http://www.pdl.cmu.edu/PDL-FTP/BigLearning/gaia.pdf the 3 state-of-the-art Distributed Machine Learning systems are

Bösen (J. Wei, W. Dai, A. Qiao, Q. Ho, H. Cui, G. R. Ganger, P. B. Gibbons, G. A. Gibson, and E. P. Xing, “Managed communication and consistency for fast data-parallel iterative analytics,” in SoCC, 2015.),

GeePS (H. Cui, H. Zhang, G. R. Ganger, P. B. Gibbons, and E. P. Xing, “GeePS: Scalable deep learning on distributed GPUs with a GPU-specialized parameter server,” in EuroSys, 2016, software available at https://github.com/cuihenggang/geeps)

and IterStore (H. Cui, A. Tumanov, J. Wei, L. Xu, W. Dai, J. Haber-Kucharsky, Q. Ho, G. R. Ganger, P. B. Gibbons, G. A. Gibson, and E. P. Xing, “Exploiting iterative-ness for parallel ML computations,” in SoCC, 2014, software available at https://github.com/cuihenggang/iterstore.)

File: Generic_implementations.tex [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+\end{figure}
+
+Furthermore the Map-reduce framework is similar to the Bulk-Synchronous processing (BSP)
+style which is a little older, however, there are some difference as the Map-reduce
+framework doesn't allow communication between nodes in the map phase, it does allow
+communication in the reduce phase when all key value pairs are passed between workers[5].
+Also, as BSP and Map-reduce are so similar work has gone into transforming BSP tasks into
+map-reduce tasks and as has been shown by Goodrich et al.[4] all BSP programs can be
+converted into Map-reduce programs. Other researchers have gone as far as to say
+that all map-reduce task are so similar to BSP tasks that, becuase BSP has a more theoretical
```
Comment (Kloppie5) [inline Generic_implementations.tex:65]: BECUASE

File: Generic_implementations.tex [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+which knows exactly where each chunk is saved.[1] The Data and workflow is described
+in figure \ref{GFS_Architecture} below.
+
+\begin{figure}
+  \includegraphics[scale=0.5]{GFS_Architecture.png}
+  \caption{Google File System architecture[1]}
+  \label{GFS_Architecture}
+\end{figure}
+
+The GFS architecture that was desribed in their paper by google, was adapeted into
```
Comment (Kloppie5) [inline Generic_implementations.tex:25]: desribed, adapted, soluttion, ditributed

**Review by Kloppie5 — APPROVED** [review]
Maybe move the references to the bib file, but otherwise lgtm

**Review by jverbraeken — APPROVED** [review]

**Pull request merged**